tests/units/test_remotes.rb

```diff
@@ -122,6 +122,34 @@ def test_fetch
       assert(loc.tags.map(&:name).include?('test-tag-in-deleted-branch'))
     end
   end
+
+  def test_fetch_ref_adds_ref_option
+    in_temp_dir do |path|
+      loc = Git.clone(@wbare, 'local')
+      rem = Git.clone(@wbare, 'remote', :config => 'receive.denyCurrentBranch=ignore')
+      loc.add_remote('testrem', rem)
+      
+      loc.chdir do
+        new_file('test-file1', 'gonnaCommitYou')
+        loc.add
+        loc.commit('master commit 1')
+        first_commit = loc.log.first.sha
+        
+        new_file('test-file2', 'gonnaCommitYouToo')
+        loc.add
+        loc.commit('master commit 2')
+        second_commit = loc.log.first.sha
+
+        # Make sure fetch message only has the first commit when we fetch the first commit
+        assert(loc.fetch('origin', {:ref => first_commit}).include?(first_commit))
```

Comment:
Any thought on how to make this a little more robust?

Reply:
I think it might be more pertinent to test that the `#command` method receives the expected argument strings, rather than testing the functionality of git from this high up.

Perhaps something like

```ruby
assert_send(loc, :command, ["fetch", "origin", first_commit])
```

Some other considerations:

Can these refs be an array? What if I want to do this: (from [git fetch's docs](https://git-scm.com/docs/git-fetch#_examples))

```
git fetch origin +pu:pu maint:tmp
```

This *could* be a future improvement.

I'm a little skeptical about using the name `first_commit`, when it's actually just the sha. It might be more clear to the reader if you call it `first_commit_sha`.

Reply:
Yeah, that's a bad name, I thought I'd use more than the sha, then I forgot to update the variable name.
I'll look into testing the `#command` side of things, I wasn't super familiar with how best to accomplish that, so your example is really helpful
Also, refs could be an array, but I didn't take that into account. I'll see if that's easy to support, which I bet it is. If it's too annoying to add, I'll at least add a comment about the limitation.

Thanks for reviewing!!!

Reply:
Ok, so adding `assert_send([loc.lib, :command, "fetch", ["origin", first_commit_sha]])` doesn't actually help test, as it passes as long as there isn't an error raised.
I think testing it from a high-level is probably #goodEnough

Normally I would just use `expect(loc.lib).to receive(...)` if I were using rspec, but it seems like UnitTest is just not very featured when it comes to mocking.